### PR TITLE
Add missing header bars and CSD

### LINKED ISSUE #26: "Transplant" necessary changes from MediterraneanNight to other themes.
To make the themes be even more similar (and all working as well as they should), we should "transplant" the changes that I made in the `MediterraneanNight` theme to the other themes. These can be seen (partially) with the command:

    git diff --ignore-all-space 60c2b714.. MediterraneanNight

### PR DESCRIPTION
**Pull request by robertpainsi**
Copied the header bars and CSD stuff from the MediterraneanNight theme
to all other themes.

See: https://github.com/rbrito/pkg-mediterranean-gtk-themes/issues/26

This doesn't close https://github.com/rbrito/pkg-mediterranean-gtk-themes/issues/26 but fixes all differences in the gtk-3.0 themes. I will add the missing changes which has to be done for gtk-2.0 to the corresponding issue.

Nevertheless... I still have some concerns about the added stuff. Variables like unfocused_borders, button_hover_gradient_color_a and some more aren't defined anywhere in the collection... So, is this a problem? If so, we have to define it in **every** theme (including MediterraneanNight).

I also **didn't test** it because I have no idea what these changes do or which parts of a window are effected :smile_cat:.

BTW: The result of comparing all gtk-widgets.css files:

```
find . -name "gtk-widgets.css" -exec echo {} \; -exec diff MediterraneanNight/gtk-3.0/gtk-widgets.css {} \;

./MediterraneanLightDarkest/gtk-3.0/gtk-widgets.css
./MediterraneanGrayDark/gtk-3.0/gtk-widgets.css
./MediterraneanTribute/gtk-3.0/gtk-widgets.css
./MediterraneanTributeDark/gtk-3.0/gtk-widgets.css
2404c2404
<     color: @theme_base_color;

---
>     color: @text_color;
./MediterraneanNight/gtk-3.0/gtk-widgets.css
./MediterraneanDark/gtk-3.0/gtk-widgets.css
2404c2404
<     color: @theme_base_color;

---
>     color: @text_color;
./MediterraneanDarkest/gtk-3.0/gtk-widgets.css
2404c2404
<     color: @theme_base_color;

---
>     color: @fg_color;
./MediterraneanLight/gtk-3.0/gtk-widgets.css
./MediterraneanNightDarkest/gtk-3.0/gtk-widgets.css
./MediterraneanWhite/gtk-3.0/gtk-widgets.css
2404c2404
<     color: @theme_base_color;

---
>     color: @text_color;
./MediterraneanWhiteNight/gtk-3.0/gtk-widgets.css
2404c2404
<     color: @theme_base_color;

---
>     color: @text_color;
./MediterraneanTributeBlue/gtk-3.0/gtk-widgets.css
```

This means, these files are almost identical! :smile_cat:
I'll try to remove the differences completely later.

### CODE REVIEW

**Comment by rbrito** [comment]
Hi, Robert.

On Feb 03 2015, Robert Painsi wrote:

> Copied the header bars and CSD stuff from the MediterraneanNight theme
> to all other themes.

Great!

> This doesn't close
> https://github.com/rbrito/pkg-mediterranean-gtk-themes/issues/26 but fixes
> all differences in the gtk-3.0 themes. I will add the missing changes
> which has to be done for gtk-2.0 to the corresponding issue.

Great. There is not so much work left, I think.

> Nevertheless... I still have some concerns about the added
> stuff. Variables like unfocused_borders, button_hover_gradient_color_a and
> some more aren't defined anywhere in the collection... So, is this a
> problem? If so, we have to define it in **every** theme (including
> MediterraneanNight).

I won't merge it right now, pending some experiments. Which theme do you
personally use? As you may have noted, I use MediterraneanNight and I only
tweaked it up to the point where I could live with it (there are some points
that I would like to have improved, but I am a bit short on time right now).

> I also **didn't test** it because I have no idea what these changes do or
> which parts of a window are effected :smile_cat:.

If I remember correctly, headerbars are the new "embedded" bars with titles
and buttons that GNOME applications like gnome-system-monitor (and many
others) use, instead of using the window manager to draw the title bar of a
window (to grab and move windows, for example).

> BTW: The result of comparing all gtk-widgets.css files:
> 
> ```
> find . -name "gtk-widgets.css" -exec echo {} \; -exec diff MediterraneanNight/gtk-3.0/gtk-widgets.css {} \;
> (...)
> This means, these files are almost identical! :smile_cat:
> ```

That's my original idea: to refactor/change the files so that they are as
similar as possible and, in the future, to have only a single file.

As a consequence, one of the ideas that I had was that the theme would be
the same for all variants and, then, the person would simply choose the
colors and a program (like a Python script) would simply generate the theme
files based on a skeleton (or, if the desktop environments allow it, just
let the user customize the colors).

In other words, find as much commonality as possible.

> I'll try to remove the differences completely later.

Great! I will wait a little bit before merging the changes, so that we can
experiment with them and see if they don't break many things (e.g., the
undefined variables).

Thanks,

## 

Rogério Brito : rbrito@{ime.usp.br,gmail.com} : GPG key 4096R/BCFCAAAA
http://cynic.cc/blog/ : github.com/rbrito : profiles.google.com/rbrito
DebianQA: http://qa.debian.org/developer.php?login=rbrito%40ime.usp.br

**Comment by robertpainsi** [comment]
All right!
Yeah, I agree, there is no need to merge it yet. I think that all undefined variables are ignored completely. Nevertheless, we have to fix this for the MediterraneanNight first because this code is already in the master :scream_cat:

> Which theme do you personally use?

MediterraneanDarkest with colored min/max/close buttons and no noisy stuff :smirk_cat:

> That's my original idea: to refactor/change the files so that they are as
> similar as possible and, in the future, to have only a single file.
> 
> As a consequence, one of the ideas that I had was that the theme would be
> the same for all variants and, then, the person would simply choose the
> colors and a program (like a Python script) would simply generate the theme
> files based on a skeleton (or, if the desktop environments allow it, just
> let the user customize the colors).
> 
> In other words, find as much commonality as possible.

Great idea! Let's see how far we can get :smirk_cat:

**Comment by robertpainsi** [comment]
Uhh!
https://github.com/rbrito/pkg-mediterranean-gtk-themes/issues/15 can be resolved by this PR :smiley_cat:

Nevertheless these changes shouldn't be merged. The MediterraneanDarkest theme has no round corners at the titlebar. With these changes it will have them and I don't like them :pouting_cat:
There are also other flaws which already have been mentioned.

**Comment by robertpainsi** [comment]
Closed because of enhanced version. Details https://github.com/rbrito/pkg-mediterranean-gtk-themes/issues/19#issuecomment-73531040

Nevertheless this doesn't solve the issue and I also have to retract all of my comments above. It should be redone by scratch.

Also some code has been changed and all undefined colors have been defined (correctly?).

https://github.com/robertpainsi/pkg-mediterranean-gtk-themes/blob/cad144ac32d3cec1cbc8d8138f44ffb7b1fc4994/MediterraneanNight/gtk-3.0/scss/gtk.scss#L43-L46
https://github.com/robertpainsi/pkg-mediterranean-gtk-themes/blob/cad144ac32d3cec1cbc8d8138f44ffb7b1fc4994/MediterraneanNight/gtk-3.0/scss/gtk-widgets.scss#L7-L11
